packages/graphback-core/src/plugin/GraphbackPluginEngine.ts

```diff
@@ -71,9 +71,11 @@ export class GraphbackPluginEngine {
   private createSchema() {
     //We need to apply all required changes to the schema we need
     //This is to ensure that every plugin can add changes to the schema
+    this.metadata.setModelDefinitions();
     for (const plugin of this.plugins) {
       const newSchema = plugin.transformSchema(this.metadata);
       this.metadata.setSchema(newSchema);
+      this.metadata.setModelDefinitions();
     }
```

Comment:
```suggestion
    for (const plugin of this.plugins) { 
      this.metadata.setModelDefinitions();
      const newSchema = plugin.transformSchema(this.metadata);
      this.metadata.setSchema(newSchema);
    }
```

Reply:
I will need to revert to how I had it first, I was seeing integration tests failing in the DataSync plugin (https://github.com/aerogear/graphback/pull/2057/checks?check_run_id=1127868628)

The reason for this is because the DataSync plugin programatically applies extra metadata to the schema, so the model definitions need to be built again _after_ the `transformSchema` hook.